File: src/cam/utils/ansi.py
"""ANSI escape sequence stripping utility."""

from __future__ import annotations

import re

# Matches common ANSI escape sequences:
#   - CSI sequences: ESC [ ... <final byte>
#   - OSC sequences: ESC ] ... BEL
#   - Other ESC sequences: ESC <intermediate byte>
_ANSI_RE = re.compile(
    r"\x1B\[[0-9;?]*[ -/]*[@-~]"   # CSI sequences (colors, cursor, etc.)
    r"|\x1B\][^\x07]*\x07"          # OSC sequences (title, etc.)
    r"|\x1B[@-_]"                    # Two-char ESC sequences
)


def strip_ansi(text: str) -> str:
    """Remove ANSI escape sequences from text.

    Args:
        text: Input text potentially containing ANSI codes.

    Returns:
        Text with all ANSI escape sequences removed.
    """
    return _ANSI_RE.sub("", text)

File: cam/cam-agent/internal/screen/ansi.go
package screen

import "regexp"

// ansiPattern matches ANSI escape sequences (CSI, OSC, and simple escapes).
var ansiPattern = regexp.MustCompile(`\x1b\[[0-9;]*[a-zA-Z]|\x1b\][^\x07\x1b]*(?:\x07|\x1b\\)|\x1b[()][AB012]|\x1b[=>]`)

// StripANSI removes all ANSI escape sequences from the input string.
func StripANSI(s string) string {
	return ansiPattern.ReplaceAllString(s, "")
}
